### DISCUSSION TITLE
Add an option to ignore emitting VisibilityNotifier siglals when the node exits the tree

### DISCUSSION THREAD

**Discussion by me2beats** [godotengine/godot-proposals#4399] [Engine Core]
Say I make 2D 🎈 balloon shooter like in old Siemens mobile.
Each balloon has a VisibilityNotifier.
I have a label that displays the count of missed balloons (so I use screen_exited or viewport_exited signal, which increments the count of missed balloons when a balloon is outside the game window.

The problem is when I shoot a balloon, the above signals also emitted, because I use balloon.queue_free() that calls tree_exited signal and in this case I don't want it.

The solution would be to have an option to ignore tree_exited signal.

**Comment by Mickeon** [comment]
When the Balloon exits the tree, it also goes out of view. It does make sense, to me. If this behaviour could be disabled, what would happen when the Node moves in and out of the tree? Should it also not be emitted on `tree_entered`? To me it's asking for a bit of inconsistency.
